api/backend/app/services/trigger_history_service.py

```diff
@@ -100,20 +100,32 @@ def format_query(self, function_name: list[str] = None, agent_id: str = None):
     def process_chart_data(self, raw_data: List[Dict[str, Any]], max_range: int) -> List[Dict[str, Any]]:
         result = [{"count": 0, "values": {}} for _ in range(max_range)]
         now = datetime.now(timezone.utc)
+
         for item in raw_data:
             timestamp_str = item.get("minute") or item.get("hour") or item.get("day")
             if timestamp_str:
                 timestamp = datetime.fromisoformat(timestamp_str).replace(tzinfo=timezone.utc)
+
                 if max_range == 60:  # Last hour
-                    index = int((now - timestamp).total_seconds() / 60) % 60
+                    seconds_diff = int((now - timestamp).total_seconds() / 60)
+                    index = seconds_diff % 60 if seconds_diff < 60 else 59
```

Comment:
use `max` function

Reply:
The logic has been refactored using MIn function.